src/editor/VideoViewer.js

```diff
@@ -0,0 +1,324 @@
+        this.$videoMarkupEl = this.$el.find("pre");
+        this.$videoData = this.$el.find(".video-data");
+
+        this.$videoEl.attr("src", _getVideoUrl(this.file));
+        this.$videoEl.one("canplay", _.bind(this._onVideoLoaded, this));
+        this.$videoEl.on("error", _.bind(console.error, console));
+
+        this.$el.find(".tag-options").on("change", "input", function(){
+            var attribute = $(this).attr("setting");
+            var isEnabled = $(this).is(":checked");
+            that.videoTagSettings[attribute] = isEnabled;
+            that.updateVideoTagMarkup(true);
+        });
+
+
+        this.updateVideoTagMarkup();
+
+        _viewers[file.fullPath] = this;
+    }
+
+    // Updates the markup used in the preview & the sample markup below
+    VideoView.prototype.updateVideoTagMarkup = function(reload){
+
+        var videoTagAttributesString = "";
+
+        for(var k in this.videoTagSettings) {
+            if(this.videoTagSettings[k]) {
+                videoTagAttributesString = videoTagAttributesString + " " + k;
+                this.$videoEl.attr(k,"");
+            } else {
+                this.$videoEl.removeAttr(k);
+            }
+        }
+
+        var videoTagMarkup =
+`<video`+ videoTagAttributesString +`>
+  <source src="`+ _getLocalVideoUrl(this.file) + `" type="video/mp4">
+</video>`;
+
+
+        this.$videoMarkupEl.text(videoTagMarkup);
+
+        // Reloads the video when one of the attributes is changed
+        // so that the preview reflects the changes
+        if(reload) {
+            this.$videoMarkupEl.one("animationend",function(){
+                $(this).removeClass("pop");
+            });
+
+            this.$videoMarkupEl.addClass("pop");
+
+            var videoWrapperHeight = this.$videoWrapperEl.height();
+
+            this.$videoWrapperEl.css("min-height", videoWrapperHeight);
+            this.$videoEl.one("canplay", _.bind(this._onVideoReloaded, this));
+            this.$videoEl[0].load();
+        }
+    }
+
+
+    /**
+     * DocumentManger.fileNameChange handler - when an video is renamed, we must
+     * update the view
+     *
+     * @param {jQuery.Event} e - event
+     * @param {!string} oldPath - the name of the file that's changing changing
+     * @param {!string} newPath - the name of the file that's changing changing
+     * @private
+     */
+    VideoView.prototype._onFilenameChange = function (e, oldPath, newPath) {
+        /*
+         * File objects are already updated when the event is triggered
+         * so we just need to see if the file has the same path as our video
+         */
+        if (this.file.fullPath === newPath) {
+            this.relPath = ProjectManager.makeProjectRelativeIfPossible(newPath);
+        }
+    };
+
+
+    /* Removes min-height property */
+    VideoView.prototype._onVideoReloaded = function (e) {
+        var videoWrapperEl = $(this.container).find(".video-wrapper");
```

Comment:
You cached this on `this` above https://github.com/mozilla/brackets/pull/807/files#diff-8ce4bff2fd0934afa9a004d220f095c7R105